tmc-langs-r/src/main/java/fi/helsinki/cs/tmc/langs/r/RPlugin.java

```diff
@@ -0,0 +1,187 @@
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+public final class RPlugin extends AbstractLanguagePlugin {
+
+    /**
+     * R folder contains the actual R files used in the 
+     * project/package. It is automatically included when creating a
+     * R package but now when making a regular project in RStudio.
+     */
+    private static final Path R_FOLDER_PATH = Paths.get("R");
+
+    /** 
+     * test/testthat folder contains the unit testing 
+     * files which use the testThat library for the R project.
+     */
+    private static final Path TEST_FOLDER_PATH = Paths.get("tests");
+    private static final Path TESTTHAT_FOLDER_PATH = Paths.get("testthat");
+
+    private static final String CANNOT_RUN_TESTS_MESSAGE = "Failed to run tests.";
+    private static final String CANNOT_PARSE_TEST_RESULTS_MESSAGE = "Failed to read test results.";
+    private static final String CANNOT_SCAN_EXERCISE_MESSAGE = "Failed to scan exercise.";
+    private static final String CANNOT_PARSE_EXERCISE_DESCRIPTION_MESSAGE =
+            "Failed to parse exercise description.";
+
+    private static Logger log = LoggerFactory.getLogger(RPlugin.class);
+
+    public RPlugin() {
+        super(
+                new ExerciseBuilder(),
+                new StudentFileAwareSubmissionProcessor(),
+                new StudentFileAwareZipper(),
+                new StudentFileAwareUnzipper());
+    }
+
+    /**
+     * NOTE: Files.exists does not seem to be able to verify the R and 
+     * testthat folder's existence if they are empty.
+     */
+    @Override
+    public boolean isExerciseTypeCorrect(Path path) {
+        return Files.exists(path.resolve(R_FOLDER_PATH))
+                || Files.exists(path.resolve(TEST_FOLDER_PATH).resolve(TESTTHAT_FOLDER_PATH));
+    }
+
+    @Override
+    protected StudentFilePolicy getStudentFilePolicy(Path projectPath) {
+        return new RStudentFilePolicy(projectPath);
+    }
+
+    @Override
+    public String getPluginName() {
+        return "r";
+    }
+
+    @Override
+    public Optional<ExerciseDesc> scanExercise(Path path, String exerciseName) {
+        ProcessRunner runner = new ProcessRunner(this.getAvailablePointsCommand(), path);
+
+        try {
+            runner.call();
+        } catch (Exception e) {
+            log.error(CANNOT_SCAN_EXERCISE_MESSAGE, e);
```

Comment:
Is the some scenario where we don't want to just early return `Optional.absent()` here?